### ISSUE TITLE
"Audio High Fidelity" Engine

### ISSUE THREAD

**Issue by Nadahar** [UniversalMediaServer/UniversalMediaServer#803]
@UniversalMediaServer/developers What exactly does this do? I've never seen it do anything before, but I'm not testing UMS with Kodi and when browsing FLAC files it does something. The thing is, whatever it does it doesn't work properly. 

For some reason, every track is listed two times (not in the TRANSCODE folder but in the "normal"). First each track appears with `<filename> [Audio High Fidelity]` and then it appears with it's tag title only (Which I would deem to be the correct thing). The first entry (the [Audio High Fidelity] one) won't play, but the second does. 

So, I figured I'll just disable this engine to get rid of the non-working duplicates. But no, when I do this EVERYTHING disappears, even the "working" entries. What's up with this, any explanation/ideas/information or do I have to spend hours figuring this out?

**Comment by Sami32** [comment]
@Nadahar Wow...you teach me that this engine is doing something :-D
i saw that engine, but didn't how to use it, i didn't search as well, but since no informations was given i though it was just for a futur implementation.
I play FLAC on Panasonic Viera but didn't have this problem. I don't know what renderer you was using, i didn't check the code but i know by experience that weird thing like that could come from PS3 trick/workaround
But after reading you, i discover that it's only used by TsMuxer for audio.
So instead of writing "No settings for now" maybe a short description of what it's used for could be more meaningful ;-D

**Comment by Sami32** [comment]
Just a verification, in case it could be related to that issue https://github.com/UniversalMediaServer/UniversalMediaServer/issues/881

Do you have again this problem with the latest UMS version ?

**Comment by Nadahar** [comment]
@Sami32 I haven't retested with the latest version, but I see now reason why it should work if noone explicitly did something to fix it. 

Whatever the intention was, It has some bugs in it listing every file twice and removing the files if you disable the engine.

**Comment by Sami32** [comment]
@Nadahar If never your test it, don't forget to give feedback or log :wink: 
Many users, like me, use it, so i'm not able to guess without some more informations.

**Comment by Nadahar** [comment]
@Sami32 I really have to stretch my imagination to see how some work on 24 bit FLAC could solve this. It's a logical bug listing them twice and hiding them if it's turned off, it won't go away without someone specificly addresses that. When it comes to why the playback doesn't work, it might or might not have been affected, but that wasn't my primary concern here.

**Comment by Sami32** [comment]
@Nadahar Your issue deal only with 24 bit FLAC as far i could understand, as it's the only purpose of this engine; you make me rember that _flac.exe_ was upgraded as well recently.
I'm like you, i really have to stretch my imagination...since i didn't experimented it myself.

**Comment by Nadahar** [comment]
I don't have a late version of UMS installed so I can't easily test it, but since Kodi was the renderer it would be easy for anyone to test.

**Comment by Sami32** [comment]
No disk place for Kodi for me, so i pass my turn.

**Comment by ExSport** [comment]
Don't believe you don't have 100MB free space without doing anything or by lowering pagefile/swap or using USB stick, anything...

**Comment by Sami32** [comment]
@Exsport Near...but you do, so i leave it you ;-)

**Comment by ExSport** [comment]
Give me sample to test and no problem. Unfortunately I don't have PanTV or PS3 in hands (only when I am with parents) so I can test it with SW renderers only.

**Comment by Nadahar** [comment]
@Sami32 @ExSport Since I had stash and pull UMS master anyway to retest #588 I decided I might as well redo this one as well. But, in the meanwhile some change has broken Kodi completely in UMS.

UMS detects Kodi and everything looks normal, but Kodi doesn't find UMS anymore. I already had saved a shortcut to UMS in Kodi, and Kodi just replies "Couldn't connect to network server". That was Kodi 15.2 which was the version I used for testing with UMS earler (and that worked then). I then upgraded to the lastest stable (16.1) to no avail, the exact same thing happens. Kodi does query UMS for contentdirectory and UMS says it answers, but something is probably wrong with the answer. Kodi renderer configuration is unchanged/standard.

Before anyone asks, I'm NOT going to build every version since 6.0.0 (which is approximately where we were when I tested this last) to see when it broke, anyone that want to find out what broke it should do that themselves.

I've attached debug files to make @Sami32 happy, although I'm betting the result isn't very hard to reproduce. 
[ums_dbg.zip.txt](https://github.com/UniversalMediaServer/UniversalMediaServer/files/453686/ums_dbg.zip.txt)
